docs/application/native/guides/text-input/multi-assistant.md

```diff
@@ -0,0 +1,444 @@
+       if (MA_ERROR_NONE != ret)
+           /* Error handling */
+   }
+   ```
+
+   `ma_prepare()` is asynchronous and when the preparation succeeds, the client state changes from `MA_STATE_INITIALIZED` to `MA_STATE_READY`. If `ma_prepare()` fails, the error callback is triggered.
+
+4. When the Multi-assistant is no longer needed, unprepare and deinitialize it:
+
+   ```c
+   void
+   unprepared_multi_assistant()
+   {
+       int ret;
+       ret = ma_unprepare();
+       if (MA_ERROR_NONE != ret)
+           /* Error handling */
+   }
+
+   void
+   deinitialize_multi_assistant()
+   {
+       int ret;
+       ret = ma_deinitialize();
+       if (MA_ERROR_NONE != result)
+           /* Error handling */
+   }
+   ```
+
+   When `ma_unprepare()` succeeds, the client state changes from `MA_STATE_READY` to `MA_STATE_INITIALIZED`.
+
+   > [!NOTE]
+   > Do not call `ma_deinitialize()` in a callback. Within a callback, `ma_deinitialize()` fails and returns `MA_ERROR_OPERATION_FAILED`.
+
+
+<a name="callback"></a>
+## Manage callbacks
+
+For more information on the callback functions, see the `multi_assistant_common.h` header file, where they are defined.
+
+To set and unset callbacks to get notifications about state changes, errors and audio data retrieval:
+
+> [!NOTE]
+> Set and unset all callbacks when the client state is `MA_STATE_INITIALIZED`.
+
+- Set the state change callback to be invoked when the client state changes:
+
+  ```c
+  /* Callback */
+  void
+  state_changed_cb(ma_state_e previous, ma_state_e current, void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_state_changed_cb()
+  {
+      int ret;
+      ret = ma_set_state_changed_cb(state_changed_cb, NULL);
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_state_changed_cb()
+  {
+      int ret;
+      ret = ma_unset_state_changed_cb();
+      if (MA_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the service state change callback to be invoked when the Multi-assistant service state changes:
+
+  User controls the Multi-assistant service state. In a general scenario:
+
+  1. If the system is configured to use the Multi-assistant feature, then after the system boots the Multi-assistant service will have the `MA_SERVICE_STATE_LISTENING` state.
+  2. User starts giving the voice command to the system by saying the wake word or pressing the voice button. If the start is successful, the Multi-assistant service state changes to `MA_SERVICE_STATE_UTTERANCE`.
+  3. After recording, the service state changes to `MA_SERVICE_STATE_PROCESSING` for recognition processing.
+  4. After the recognition is completed, the service state returns to `MA_SERVICE_STATE_LISTENING`.
+
+  If the system configuration changes and because of this if the Multi-assistant feature is not available, then the Multi-assistant service state can be changed to `MA_SERVICE_STATE_INACTIVE`.
```

Comment:
Well the reason why I didn't put this description in the overview section is that, when developing a voice assistant, in most cases, this information is not something the application is interested in. In some situations this can be used for a specific purpose, but in general it is not very informative to the assistant application. Do you think it'd be better to put this in the overview section even in this case? I would appreciate if you could provide your suggestion on this.

Reply:
Thanks for the answer @dalton81.
The reason I asked to move this description to overview is that, 
I thought this state change scenario as a kind of general information.

This description would not be important, but it seems basic background information about the topic. We provide the basic simple information about the module in the overview section, so I think it would be good to move this description to the overview.

Would you please let me know if you still want to keep this description at here? It would be helpful if you let us know your opinion to my answer.